backends-velox/src/main/scala/io/glutenproject/utils/ArrowAbiUtil.scala

```diff
@@ -79,8 +137,16 @@ object ArrowAbiUtil {
         .map(v => v)
     new ColumnarBatch(vectors, rowCount)
   }
+  // will release input record batch
+  private def toVectorSchemaRoot(allocator: BufferAllocator, batch: ColumnarBatch)
+  : VectorSchemaRoot = {
+    toVectorSchemaRoot(allocator, batch, null)
+  }
 
-  private def toVectorSchemaRoot(batch: ColumnarBatch): VectorSchemaRoot = {
+  // will release input record batch
+  private def toVectorSchemaRoot(allocator: BufferAllocator, batch: ColumnarBatch,
```

Comment:
What do you think about using a separated method like `toVectorSchemaRoot(BufferAllocator, ArrowRecordBatch)`?

Reply:
Because in shuffle code, we need to estimate the data size, it need to get from `ArrowRecordBatch`
val rb = ArrowConverterUtils.createArrowRecordBatch(cb) 
ArrowAbiUtil.exportFromSparkColumnarBatch(allocator, cb, null, cArray, rb) `dep.dataSize.add(rb.getBuffersLayout.asScala.map(buf => buf.getSize).sum)`